src/ws/login.js

```diff
@@ -614,13 +614,21 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
     }
 
     function login_reload (wanted) {
+        // Force a reload if not triggered below
+        var timer = window.setTimeout(function() {
+            timer = null;
+            window.location.reload(true);
+        }, 100);
+
         if (wanted && wanted != window.location.href)
             window.location = wanted;
```

Comment:
Here we seem to change the location, but above we have reloaded. Wouldn't we do either or?

Reply:
This is the way it was before too, except that if the new page didn't load in time the callback would fire. I think it was there to ensure the load would happen but I don't remember under what conditions it might be needed. But if we don't think we need it anymore then I'm ok getting rid of the window.location.reload callback and just doing it in a else clause.

Reply:
Yeah, seems like we could try that first.

Reply:
@stefwalter so I remembered what this for. If wanted and current location are only differ by a hash then setting location won't trigger a page reload. So the timeout will kick in and trigger it.